### PR TITLE
cocoapi can be installed as a catkin package now

### PR DESCRIPTION
**Pull request by abhijitmajumdar**

### CODE REVIEW

File: README.txt [hunk 1]
```diff
@@ -13,3 +13,16 @@ To install:
 -For Matlab, add coco/MatlabApi to the Matlab path (OSX/Linux binaries provided)
 -For Python, run "make" under coco/PythonAPI
 -For Lua, run “luarocks make LuaAPI/rocks/coco-scm-1.rockspec” under coco/
+-For c++ ROS catkin package, see below
+
+Build and usage with ROS packages (using catkin as an example):
+- Assuming your sources are located in "~/catkin_ws/src", add this repository to your sources: "git clone https://github.com/abhijitmajumdar/cocoapi.git ~/catkin_ws/src/cocoapi"
```
Comment (aaronplusone) [inline README.txt]: I would omit this and instead say "Add to your workspace".  The path will change and those in ROS land will understand.

File: package.xml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0"?>
+<package format="3">
+    <name>cocoapi</name>
+    <version>0.1.0</version>
+    <description>A ROS c++ package comiling the cocoAPI</description>
+    <maintainer email="developers@plusone-robotics.com">Plus One Robotics, Inc.</maintainer>
+    <author>Piotr Dollar</author>
+    <author>Tsung-Yi Lin</author>
+    <license>Proprietary</license>
```
Comment (aaronplusone) [inline package.xml]: I think we have to leave the license as it is. In github, it looks like a BSD-2-Clause

File: package.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0"?>
+<package format="3">
+    <name>cocoapi</name>
+    <version>0.1.0</version>
+    <description>A ROS c++ package comiling the cocoAPI</description>
```
Comment (aaronplusone) [inline package.xml]:
```suggestion
    <description>A ROS c++ package for cocoAPI</description>
```

File: README.txt [hunk 4]
```diff
@@ -13,3 +13,16 @@ To install:
...
+- For a package that needs to use "cocoapi" (say "user_pkg") assign cocoapi as a dependency:
+  - set "<depend>cocoapi</depend>" in the "package.xml" for the "user_pkg"
+  - in your "CMakeLists.txt" for "user_pkg" set "find_package(catkin REQUIRED COMPONENTS cocoapi)"
+  - in your "CMakeLists.txt" for "user_pkg" set "catkin_package(... CATKIN_DEPENDS cocoapi)"
+  - in your "CMakeLists.txt" for "user_pkg" setting "include_directories(${catkin_INCLUDE_DIRS})" allows finding the headers
+  - in your "CMakeLists.txt" for "user_pkg" setting "target_link_libraries(... ${catkin_LIBRARIES})" allows linking the shared library for "cocoapi"
+- Once you build your package, it should build cocoapi because of the dependency tree
```
Comment (aaronplusone) [inline README.txt]: Omit this as well.

**Review by aaronplusone — CHANGES_REQUESTED** [review]
License needs to be addressed - @shaun-edwards, care to comment? Other review feedback is minor.

File: package.xml (hunk 2 above)
Comment (shaun-edwards) [reply package.xml]: @aaronplusone is correct.  Let's just add it here for completeness.  The license itself is OK to use.

Comment (abhijitmajumdar) [reply package.xml]: I replaced it with `<license>BSD-2-Clause</license>`. Does that work?

**Review by aaronplusone — APPROVED** [review]

**Pull request merged**